atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java

```diff
@@ -131,16 +132,25 @@ private void clearLockedCell(CellReference cellReference) {
                 .orElse(map));
     }
 
-    private Stream<CellReference> extractCandidateCells(LockDescriptor descriptor) {
-        return AtlasLockDescriptorUtils.candidateCells(descriptor).stream();
+    private void applyLockedDescriptors(Set<LockDescriptor> lockDescriptors) {
+        getCandidateCells(lockDescriptors).forEach(this::putLockedCell);
     }
 
-    private void applyLockedDescriptors(Set<LockDescriptor> lockDescriptors) {
-        lockDescriptors.stream().flatMap(this::extractCandidateCells).forEach(this::putLockedCell);
+    private Stream<CellReference> getCandidateCells(Set<LockDescriptor> lockDescriptors) {
+        return lockDescriptors.stream()
+                .map(AtlasLockDescriptorUtils::tryParseTableRef)
+                .flatMap(Optional::stream)
+                // Explicitly exclude descriptors corresponding to watched rows from non-watched tables
+                .filter(ValueStoreImpl.this::isTableWatched)
+                .flatMap(ValueStoreImpl.this::extractCandidateCells);
```

Comment:
nit: I don't think you need the `ValueStoreImpl` before `this` as there aren't nested classes in this bit (I expect this came from IntelliJ)